send_to_dock/test/test_send_to_dock_node.cpp

```diff
@@ -0,0 +1,131 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+
+#include <nav2_msgs/action/dock_robot.hpp>
+
+#include "send_to_dock/send_to_dock_node.hpp"
+
+using DockRobot = nav2_msgs::action::DockRobot;
+using ClientGoalHandleDockRobot = rclcpp_action::ClientGoalHandle<DockRobot>;
+using ServerGoalHandleDockRobot = rclcpp_action::ServerGoalHandle<DockRobot>;
+using SetBoolSrv = std_srvs::srv::SetBool;
+
+class SendToDockNodeWrapper : public send_to_dock::SendToDockNode {
+public:
+  SendToDockNodeWrapper() {};
+
+  void HandleService(const std::shared_ptr<SetBoolSrv::Request> request,
+                     std::shared_ptr<SetBoolSrv::Response> response) {
+    return send_to_dock::SendToDockNode::HandleService(request, response);
+  }
+
+  void SetActiveGoal() {
+    send_to_dock::SendToDockNode::active_goal_ =
+        std::shared_ptr<ClientGoalHandleDockRobot>(
+            reinterpret_cast<ClientGoalHandleDockRobot *>(0x1), [](auto *) {});
+  }
+};
+
+class TestSendToDockNode : public testing::Test {
+protected:
+  TestSendToDockNode();
+  rclcpp_action::Server<DockRobot>::SharedPtr CreateDockServer();
+  std::shared_ptr<SendToDockNodeWrapper> node_;
+  std::shared_ptr<SetBoolSrv::Request> request_;
+  std::shared_ptr<SetBoolSrv::Response> response_;
+  rclcpp_action::Server<DockRobot>::SharedPtr action_server_;
+};
+
+TestSendToDockNode::TestSendToDockNode() {
+  node_ = std::make_shared<SendToDockNodeWrapper>();
+  request_ = std::make_shared<SetBoolSrv::Request>();
+  response_ = std::make_shared<SetBoolSrv::Response>();
+  action_server_ = nullptr;
+}
+
+rclcpp_action::Server<DockRobot>::SharedPtr
+TestSendToDockNode::CreateDockServer() {
+  return rclcpp_action::create_server<DockRobot>(
+      node_, "dock_robot",
+      // handle_goal
+      [](const rclcpp_action::GoalUUID &,
+         std::shared_ptr<const DockRobot::Goal>) {
+        return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+      },
+      // handle_cancel
+      [](const std::shared_ptr<ServerGoalHandleDockRobot>) {
+        return rclcpp_action::CancelResponse::ACCEPT;
+      },
+      // handle_accepted
+      [](const std::shared_ptr<ServerGoalHandleDockRobot> goal_handle) {
+        auto result = std::make_shared<DockRobot::Result>();
+        goal_handle->succeed(result);
+      });
+}
+
+TEST_F(TestSendToDockNode, DockServerIsNotCreated) {
```

Comment:
I think the name of this test is misleading because this node does not create a dock server. What do you think about this?
```suggestion  
TEST_F(TestSendToDockNode, DockRobotActionServerNotAvailable) {
```